internal/nozzle/workers.go

```diff
@@ -77,9 +77,17 @@ func (d *Nozzle) readProcessedMetrics() {
 	}
 }
 
-func (d *Nozzle) handleMessage(envelope *events.Envelope) {
-	if envelope.GetEventType() == events.Envelope_CounterEvent && envelope.CounterEvent.GetName() == "TruncatingBuffer.DroppedMessages" && envelope.GetOrigin() == "doppler" {
-		d.log.Infof("We've intercepted an upstream message which indicates that the nozzle or the TrafficController is not keeping up. Please try scaling up the nozzle.")
-		d.AlertSlowConsumerError()
+func (d *Nozzle) handleMessage(envelope *loggregator_v2.Envelope) {
+	switch envelope.GetMessage().(type) {
+	case *loggregator_v2.Envelope_Counter:
+		// TODO: make sure this is correct (docs at https://docs.pivotal.io/platform/2-7/monitoring/key-cap-scaling.html)
+		if envelope.GetCounter().GetName() == "dropped" {
+			origin, ook := envelope.GetTags()["origin"]
```

Comment:
Should we have a util logic to get individual tags? I am not specifically opinionated about this.

Reply:
What would this logic look like? It's not guaranteed that the tags will be there, so we would still need to have something that would return `(tagValue string, isPresent bool)`, which doesn't feel like it's simplifying anything. Is there a better solution that I don't see?

Reply:
after a second pass, let's drop this comment.